packages/openapi3/test/metadata.test.ts

```diff
@@ -2,6 +2,29 @@ import { deepStrictEqual } from "assert";
 import { openApiFor } from "./test-host.js";
 
 describe("openapi3: metadata", () => {
+  it("can make properties optional", async () => {
+    const res = await openApiFor(`
+      model Widget { 
+        name: string; 
+        specs: WidgetSpecs;
```

Comment:
If my last comment is right, then we should add something like `subWidgets: Widget[]` and expect that the subWidgets do not have their properties made optional.

Reply:
Hmmm.  Check out the change I just pushed.  How would we prevent the `PersonRelativeUpdateItem` (and the other two synthesized schemas for that matter) from being created?

https://github.com/microsoft/cadl/pull/1345/commits/8d83adc6a42eb80ac40259fd0c4085a9d04af9d6

Reply:
See https://github.com/microsoft/cadl/pull/1345#discussion_r1039973385

Reply:
I think it would be worth still adding some unit test coverage for array case. To avoid littering the test, you can an array property that is not of a modified element type: like `parts: WidgetPart[]` where `WidgetPart` does not have any metadata visibility or reference back to Widget. It would still be good to have unit test coverage that we do not make optional properties on array elements.